src/util/props.js

```diff
@@ -0,0 +1,46 @@
+
+import { warn } from './warn'
+
+export function resolveProps (route, component, config) {
+  switch (typeof config) {
+
+    case 'undefined':
+      return
+
+    case 'object':
+      return config
+
+    case 'function':
+      return config(route)
+
+    case 'boolean':
+      if (!config) {
+        return
+      }
+      if (!component.props) {
+        return
+      }
+      const props = {}
+      for (const prop in route.params) {
+        const value = route.params[prop]
+        if (hasProp(component, prop)) {
+          const propType = component.props[prop].type
```

Comment:
I'm not sure about reading prop types at runtime and converting them to `Number` if possible.. Maybe we should just leave them as string since url params are string by nature.

Also, if this type conversion were to be supported, `type` can be an array, so we'll have to loop over that should it be the case.